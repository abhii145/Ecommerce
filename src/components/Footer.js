import React from "react";
import { FaTwitter, FaFacebook, FaInstagram } from "react-icons/fa";
import { BiStore } from "react-icons/bi";

const Footer = ({ isFixed }) => {
  const currentYear = new Date().getFullYear();

  const footerClasses = isFixed
    ? "bg-gray-900 text-white py-6 bottom-0 w-full fixed"
    : "bg-gray-900 text-white py-6 w-full";

  return (
    <footer className={footerClasses}>
      <div className="container mx-auto flex justify-between items-center">
        <BiStore size={30} className="text-gray-300 hover:text-emerald-700" />
        <div className="flex space-x-4">
          <FaTwitter size={25} className="text-gray-300 hover:text-cyan-400" />

          <FaFacebook size={25} className="text-gray-300 hover:text-sky-500" />

          <FaInstagram
            size={25}
            className="text-gray-300 hover:text-rose-400"
          />
        </div>
      </div>
      <div className="mt-2 text-center text-gray-600">
        &copy; {currentYear} Bazar. All rights reserved.
      </div>
    </footer>
  );
};

export default Footer;
