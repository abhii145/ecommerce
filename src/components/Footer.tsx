import { memo } from 'react';

const Footer = () => (
  <footer className="bg-gray-800 text-white py-6 mt-auto" role="contentinfo">
    <div className="container mx-auto flex flex-col md:flex-row justify-between items-center">
      <p className="text-sm">&copy; {new Date().getFullYear()} ShopEasy. All rights reserved.</p>
      <nav className="space-x-4 mt-4 md:mt-0" aria-label="Footer navigation">
        <a href="/" className="hover:underline" aria-label="Home">
          Home
        </a>
        <a href="/products" className="hover:underline" aria-label="Products">
          Products
        </a>
        <a href="/favourites" className="hover:underline" aria-label="Favourites">
          Favourites
        </a>
        <a href="/cart" className="hover:underline" aria-label="Cart">
          Cart
        </a>
      </nav>
    </div>
  </footer>
);

export default memo(Footer);
