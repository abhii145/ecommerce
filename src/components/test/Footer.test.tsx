import { render, screen } from "@testing-library/react"
import Footer from "../Footer"

describe("Footer component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Footer />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders Footer component", () => {
    render(<Footer />)

    expect(screen.getByText(/Â© 2024 ShopEasy/i)).toBeInTheDocument()
    expect(screen.getByText(/Home/i)).toBeInTheDocument()
    expect(screen.getByText(/Products/i)).toBeInTheDocument()
    expect(screen.getByText(/Favourites/i)).toBeInTheDocument()
    expect(screen.getByText(/Cart/i)).toBeInTheDocument()
  })
})
