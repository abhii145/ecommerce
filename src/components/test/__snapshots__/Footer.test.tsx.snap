// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="bg-gray-800 text-white py-6 mt-auto"
    role="contentinfo"
  >
    <div
      class="container mx-auto flex flex-col md:flex-row justify-between items-center"
    >
      <p
        class="text-sm"
      >
        Â© 2024 ShopEasy. All rights reserved.
      </p>
      <nav
        aria-label="Footer navigation"
        class="space-x-4 mt-4 md:mt-0"
      >
        <a
          aria-label="Home"
          class="hover:underline"
          href="/"
        >
          Home
        </a>
        <a
          aria-label="Products"
          class="hover:underline"
          href="/products"
        >
          Products
        </a>
        <a
          aria-label="Favourites"
          class="hover:underline"
          href="/favourites"
        >
          Favourites
        </a>
        <a
          aria-label="Cart"
          class="hover:underline"
          href="/cart"
        >
          Cart
        </a>
      </nav>
    </div>
  </footer>
</DocumentFragment>
`;
